main.cpp

```diff
@@ -190,285 +205,320 @@ struct Safeway
         float amountOfLastOrder = 67.59f;
         int registrationDays = 133;
         void redeemGiftCard(int codeOfGiftcard = 133278645);
-        void redeemProduct(int pointsOfProduct = 700);
+        bool redeemProduct(int pointsOfProduct = 700);
         void changeProfile(int dateOfBirthday = 13);
     };
-    //1)order grocery online
     void orderGroceryOnline(Customer customer);
-    //2)refund
     double refund(int productNumber);
-    //3)collect coupons
     void collectCoupons(Customer customer);
 };
-/*
-3) Shelter of cats
-5 properties:
-    1)number of cats
-    2)distance 
-    3)amount of mony to adopt a cat
-    4)age of a cat
-    5)activity level
-3 things it can do
-    1)adopt a cat
-    2)donate money
-    3)visit the shelter on site
- */
-struct ShelterOfCats
+
+void Safeway::orderGroceryOnline(Customer customer)
+{
+    std::cout << "Customer's profile: " << customer.accountnumber << std::endl;
+}
+
+double Safeway::refund(int productNumber)
+{
+    std::cout << "Product Number: " << productNumber << std::endl;
+    std::cout << "Product Value: 1" << std::endl;
+    return 0.1;
+}
+
+void Safeway::collectCoupons(Customer customer)
+{
+    std::cout << "Coupons " << customer.membership << std::endl;
+}
+
+void Safeway::Customer::redeemGiftCard(int codeOfGiftcard)
+{
+    std::cout << "submit " << codeOfGiftcard << std::endl;
+}
+
+bool Safeway::Customer::redeemProduct(int pointsOfProduct)
+{
+    if (pointsOfProduct > 700)
+    {
+        return false;
+    }
+    else
+    {
+        return true;
+    }
+}
```

Comment:
If you find yourself writing code like this:
```
int Type::someFunction()
{
    if( someExpression )
    {
        someStatement();
        return aValue;
    } 
    else 
    {
        return anotherValue;
    }
}
```
When the `if()` block is entered, the `return aValue;` is the last line of code to be executed.
you have no code after the `else` block that needs executing, since your `else` block contains a `return` statement.
`return` statements are the last thing that **ever happens** in a function
That means if your `if()` block is **NOT** entered, it will be skipped.  What is the next thing that happens?  `return anotherValue;`
that means you do **NOT** need the `else` block, and it is just causing visual clutter.
```
int Type::someFunction()
{
    if( someExpression )
    {
        someStatement();
        return aValue;
    } 
    return anotherValue;
}
```
- [ ] revise